javalanglib/src/main/scala/java/lang/System.scala

```diff
@@ -6,7 +6,35 @@ import scala.scalajs.js
 import js.Dynamic.global
 import scala.scalajs.LinkingInfo.assumingES6
 
+import java.{util => ju}
+import scalajs.runtime.environmentInfo
+
 object System {
+
+  private[this] var systemProperties = loadSystemProperties()
```

Comment:
This is bad for code size, as any `println` call will cause this code to be reachable.

You'll have to use a "manual" lazy initialization, here:
```scala
private[this] var systemProperties: ju.Properties = _

def getProperties(): ju.Properties = {
  if (systemProperties == null)
    systemProperties = loadSystemProperties()
  systemPropertis
}
```
Alternatively, you can put this `var` in an `object` of its own (similarly to `IDHashCode`)